octodns/provider/base.py

```diff
@@ -64,7 +64,8 @@ def plan(self, desired):
             self.log.info('plan:   filtered out %s changes', before - after)
 
         # allow the provider to add extra changes it needs
-        extra = self._extra_changes(existing, changes)
+        extra = self._extra_changes(existing=existing, desired=desired,
+                                    changes=changes)
```

Comment:
This change allows providers to catch them by name rather than position and thus ignore the params they don't want with `**kwargs`.